src/Features/Core/Portable/IncrementalCaches/MetadataInfo.cs

```diff
@@ -21,11 +21,10 @@ internal partial class SymbolTreeInfoIncrementalAnalyzerProvider
             /// <summary>
             /// The set of projects that are referencing this metadata-index.  When this becomes empty we can dump the
             /// index from memory.
-            /// <para/>
-            /// Note: the Incremental-Analyzer infrastructure guarantees that it will call all the
-            /// methods on <see cref="SymbolTreeInfoIncrementalAnalyzer"/> in a serial fashion.  As that is the only
-            /// type that reads/writes these <see cref="MetadataInfo"/> objects, we don't need to lock this.
```

Comment:
📝 This claim was added in a5973062, but has been wrong in OOP scenarios since 47d42f0. As best I can tell, bugs were never observed because we serialized metadata by references and didn't directly construct them in OOP code, so there was no way to end up with two projects sharing a metadata reference. This pull request resulted in OOP implementation code (which is running in-process for testing) sharing some metadata references, so we have to correct this at the same time.